drafts/building-github-pull-requests-with-jenkins-pipelines.md

```diff
@@ -0,0 +1,114 @@
+# Building Github Pull Requests with Jenkins Pipelines
+
+We've written about building Github Pull Requests using Jenkins [before](/building-github-pull-requests-with-jenkins/), back in 2013. Now in 2016 we're still strong Jenkins users here at Kabisa and Jenkins has changed quite a bit in recent years. Most notably Jenkins  introduced the notion of [Pipelines](https://jenkins.io/solutions/pipeline/).
+
+Jenkins Pipelines allow you to describe your entire CI/CD pipeline in code and version it together with your production code. Additionally the whole Jenkins Pipelines ecosystem allows for many powerful setups, including [building jobs in Docker containers](https://wiki.jenkins-ci.org/display/JENKINS/CloudBees+Docker+Pipeline+Plugin) and inbuilt support for building Github Pull Requests.
+
+In this post I'm going to show you how to configure Jenkins 2.0 with Pipelines to setup a complete CI/CD pipeline with builds running in Docker containers for isolation and ease of environment setup.
+
+## Setup Jenkins
+
+Technically Jenkins Pipelines have been available in Jenkins 1.x already, but since Jenkins 2.0 Pipelines are bundled with Jenkins itself. This means that the only prerequisite for getting started with Jenkins Pipelines is installing Jenkins 2.0!
+
+For the remainder of this post I'm going to assume you have:
+
+* Some basic knowledge of Jenkins
+* A running Jenkins 2.x instance
+* A GitHub organisation
+* A GitHub user in said organisation that has API access to list and clone repositories (either via SSH or HTTPS), as well as administer your organisation (used to configure webhooks)
+
+After following this tutorial you'll end up with the following job structure in Jenkins:
+
+```
+Github
+|--- Project 1
+  |- master
+  |- PR 1
+  |- PR 2
+  |- ...
+|--- Project 2
+  |- master
+  |- ...
+```
+
+All GitHub projects will be nested under a GitHub namespace. Jenkins will build all master branches and any Pull Requests.
+
+To get started navigate to the `New Item` page in Jenkins and configure as follows:
+
+![](https://kabisa-website-assets.s3.amazonaws.com/theguild/posts/building-github-pull-requests-with-jenkins-pipeline-style/create-organisation-directory.png)
+
+On the next screen some details about your GitHub origanisation have to be configured:
+
+![](https://kabisa-website-assets.s3.amazonaws.com/theguild/posts/building-github-pull-requests-with-jenkins-pipeline-style/configure-github-organisation.png)
+
+* **Owner**: enter the name of your GitHub organisation here
+* **Scan credentials**: select or create credentials that should be used to communicate with the GitHub API. You should use a GitHub OAuth token here that has permissions to access all repositories you'd like to build in Jenkins.
+* **Checkout credentials**: Configures which credentials should be used to clone repositories. By default https will be used using the same OAuth token used for scanning but you can also chose to use an SSH key here.
```

Comment:
...By default https will be used with the same OAuth token used for scanning; Alternatively you can use SSH.